$NetBSD: patch-configure.ac,v 1.12 2020/03/19 06:26:26 rillig Exp $

Add KERNEL_NETBSD, add entry and disk, nfs plugins for NetBSD,
and also do swap for "have_swapctl_three_args" (as for NetBSD).
Also, the start of process support, but that's not yet finished.
Correct tokyotyrant detection (add lib, not include, for -L).
Correct libupsclient detection (argument may be a searchable directory).

--- configure.ac.orig	2017-11-18 09:03:27.330751467 +0000
+++ configure.ac
@@ -110,6 +110,7 @@
 AM_CONDITIONAL([BUILD_FREEBSD], [test "x$ac_system" = "xFreeBSD"])
 AM_CONDITIONAL([BUILD_LINUX], [test "x$ac_system" = "xLinux"])
 AM_CONDITIONAL([BUILD_OPENBSD], [test "x$ac_system" = "xOpenBSD"])
+AM_CONDITIONAL([BUILD_NETBSD], [test "x$ac_system" = "xNetBSD"])
 AM_CONDITIONAL([BUILD_SOLARIS], [test "x$ac_system" = "xSolaris"])
 
 if test "x$ac_system" = "xSolaris"; then
@@ -5209,7 +5210,7 @@
       with_libtokyotyrant="$withval"
     else
       with_libtokyotyrant_cppflags="-I$withval/include"
-      with_libtokyotyrant_ldflags="-L$withval/include"
+      with_libtokyotyrant_ldflags="-L$withval/lib"
       with_libtokyotyrant_libs="-ltokyotyrant"
       with_libtokyotyrant="yes"
     fi
@@ -5328,7 +5329,7 @@
     else if test "x$withval" = "xyes"; then
       with_libupsclient="use_pkgconfig"
     else
-      if test -x "$withval"; then
+      if test -f "$withval" && test -x "$withval"; then
         with_libupsclient_config="$withval"
         with_libupsclient="use_libupsclient_config"
       else if test -x "$withval/bin/libupsclient-config"; then
@@ -6136,6 +6137,7 @@
 plugin_mcelog="no"
 plugin_memory="no"
 plugin_multimeter="no"
+plugin_netstat_udp="no"
 plugin_nfs="no"
 plugin_numa="no"
 plugin_ovs_events="no"
@@ -6228,7 +6230,9 @@
   plugin_disk="yes"
   plugin_entropy="yes"
   plugin_irq="yes"
+  plugin_nfs="yes"
   plugin_processes="yes"
+  plugin_netstat_udp="yes"
 fi
 
 # Mac OS X devices
@@ -6568,6 +6572,7 @@
 AC_PLUGIN([mysql],               [$with_libmysql],          [MySQL statistics])
 AC_PLUGIN([netapp],              [$with_libnetapp],         [NetApp plugin])
 AC_PLUGIN([netlink],             [$with_libmnl],            [Enhanced Linux network statistics])
+AC_PLUGIN([netstat_udp],         [$plugin_netstat_udp],     [UDP network statistics])
 AC_PLUGIN([network],             [yes],                     [Network communication plugin])
 AC_PLUGIN([nfs],                 [$plugin_nfs],             [NFS statistics])
 AC_PLUGIN([nginx],               [$with_libcurl],           [nginx statistics])
@@ -6987,6 +6992,7 @@
 AC_MSG_RESULT([    mysql . . . . . . . . $enable_mysql])
 AC_MSG_RESULT([    netapp  . . . . . . . $enable_netapp])
 AC_MSG_RESULT([    netlink . . . . . . . $enable_netlink])
+AC_MSG_RESULT([    netstat_udp . . . . . $enable_netstat_udp])
 AC_MSG_RESULT([    network . . . . . . . $enable_network])
 AC_MSG_RESULT([    nfs . . . . . . . . . $enable_nfs])
 AC_MSG_RESULT([    nginx . . . . . . . . $enable_nginx])
